src/snowflake/snowpark/modin/plugin/_internal/ordered_dataframe.py

```diff
@@ -612,6 +611,7 @@ def select(
         new_snowpark_column_objects: list[Column] = []
         # a list of snowflake quoted identifiers as projected columns for new OrderedDataFrame
         new_projected_columns: list[str] = []
+        active_columns = self._get_active_column_snowflake_quoted_identifiers()
```

Comment:
Do you have any profile / perf numbers on how much this improved the runtime?  Were we able to get the earlier repro (that surfaced this) to complete in reasonable time?

Reply:
Yes, it is in the description. In short, it reduces from around 16 h to 8.002492904663086 s

Reply:
Oh, sorry, you are just talking about this line,  so each check iteration is taken about 0.015 s, half of the time it is trying to getting the active columns, so each iteration is not slow, but if there are too many iterations, it becomes large.